packages/compass-collection/src/modules/collection-tab.ts

```diff
@@ -0,0 +1,244 @@
+import type { Reducer, AnyAction } from 'redux';
+import type { CollectionMetadata } from 'mongodb-collection-model';
+import type Collection from 'mongodb-collection-model';
+import type { ThunkAction } from 'redux-thunk';
+import type AppRegistry from 'hadron-app-registry';
+import type { DataService } from 'mongodb-data-service';
+import toNs from 'mongodb-ns';
+import preferencesAccess from 'compass-preferences-model';
+import React from 'react';
+
+type CollectionThunkAction<
+  ReturnType,
+  Action extends AnyAction = AnyAction
+> = ThunkAction<
+  ReturnType,
+  CollectionState,
+  {
+    globalAppRegistry: AppRegistry;
+    localAppRegistry: AppRegistry;
+    dataService: DataService;
+  },
+  Action
+>;
+
+export type CollectionStateMetadata = CollectionMetadata & {
+  serverVersion: string;
+  isAtlas: boolean;
+  isDataLake: boolean;
+};
+
+export type CollectionState = {
+  stats: Pick<
+    Collection,
+    | 'document_count'
+    | 'index_count'
+    | 'index_size'
+    | 'status'
+    | 'avg_document_size'
+    | 'storage_size'
+    | 'free_storage_size'
+  > | null;
+  metadata: CollectionStateMetadata;
+  currentTab:
+    | 'Documents'
+    | 'Aggregations'
+    | 'Schema'
+    | 'Indexes'
+    | 'Validation';
+  initialQuery?: unknown;
+  initialAggregation?: unknown;
+  editViewName?: string;
+};
+
+const initialMetadata: CollectionStateMetadata = {
+  namespace: '',
+  isReadonly: false,
+  isTimeSeries: false,
+  isClustered: false,
+  isFLE: false,
+  isSearchIndexesSupported: false,
+  isAtlas: false,
+  isDataLake: false,
+  serverVersion: '0.0.0',
+};
+
+enum CollectionActions {
+  CollectionStatsFetched = 'compass-collection/CollectionStatsFetched',
+  CollectionMetadataFetched = 'compass-collection/CollectionMetadataFetched',
+  ChangeTab = 'compass-collection/ChangeTab',
+}
+
+const reducer: Reducer<CollectionState> = (
+  state = {
+    stats: null,
+    metadata: initialMetadata,
+    currentTab: 'Documents',
+  },
+  action
+) => {
+  if (action.type === CollectionActions.CollectionStatsFetched) {
+    return {
+      ...state,
+      // TODO: pick props
```

Comment:
☝️

Reply:
Done in https://github.com/mongodb-js/compass/pull/4858/commits/21262a93395016eabbf2188904d929dcf8df4b61